### PR TITLE
tgui: Portable Turret

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR converts Portable Turret from HTML based interface to tgui, with some improvements and cleanup of the related code plus documentation of the turret's variables. Comparison images below.

<details>
  <summary>Old UI</summary>

TurretOLD

</details>

**NEW:**

TurretNEW

## Why It's Good For The Game

Better UI.

## Changelog
:cl: Arkatos
add: Portable Turret now uses tgui.
/:cl:

### CODE REVIEW

File: code/game/machinery/porta_turret/portable_turret_cover.dm [hunk 1]
```diff
@@ -29,13 +29,23 @@
 		return
 	return parent_turret.attack_ai(user)
 
+/obj/machinery/porta_turret_cover/attack_robot(mob/user)
+	. = ..()
+	if(.)
+		return
+	return parent_turret.attack_robot(user)
```
Comment (bobbahbrown) [inline code/game/machinery/porta_turret/portable_turret_cover.dm:36]:
```suggestion
	return ..() || parent_turret.attack_robot(user)
```

File: code/game/machinery/porta_turret/portable_turret_cover.dm [hunk 2]
```diff
@@ -29,13 +29,23 @@
...
 	. = ..()
 	if(.)
 		return
 	return parent_turret.attack_hand(user)
 
+/obj/machinery/porta_turret_cover/attack_ghost(mob/user)
+	. = ..()
+	if(.)
+		return
+	return parent_turret.attack_ghost(user)
```
Comment (bobbahbrown) [inline code/game/machinery/porta_turret/portable_turret_cover.dm:48]:
```suggestion
	return ..() || parent_turret.attack_ghost(user)
```

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 3]
```diff
@@ -177,68 +192,79 @@
...
-		dat += "Check for Weapon Authorization: <A href='?src=[REF(src)];operation=authweapon'>[turret_flags & TURRET_FLAG_AUTH_WEAPONS ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize Wanted Criminals: <A href='?src=[REF(src)];operation=shootcriminals'>[turret_flags & TURRET_FLAG_SHOOT_CRIMINALS ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize All Non-Security and Non-Command Personnel: <A href='?src=[REF(src)];operation=shootall'>[turret_flags & TURRET_FLAG_SHOOT_ALL ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize All Unidentified Life Signs: <A href='?src=[REF(src)];operation=checkxenos'>[turret_flags & TURRET_FLAG_SHOOT_ANOMALOUS ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize All Non-Mindshielded Personnel: <A href='?src=[REF(src)];operation=checkloyal'>[turret_flags & TURRET_FLAG_SHOOT_UNSHIELDED ? "Yes" : "No"]</A><BR>"
-		dat += "Neutralize All Cyborgs: <A href='?src=[REF(src)];operation=shootborgs'>[turret_flags & TURRET_FLAG_SHOOT_BORGS ? "Yes" : "No"]</A><BR>"
-		dat += "Ignore Heads Of Staff: <A href='?src=[REF(src)];operation=shootheads'>[turret_flags & TURRET_FLAG_SHOOT_HEADS ? "No" : "Yes"]</A><BR>"
+/obj/machinery/porta_turret/ui_data(mob/user)
+	var/list/data = list()
```
Comment (bobbahbrown) [inline code/game/machinery/porta_turret/portable_turret.dm:203]: if you cared to you could inline the assignment of all those variables inside the list, aside from the complex assignment for the silicon stuff

ie
```dm
var/list/data = (
    "locked" = locked,
    "on" = on,
    ...
)

Comment (stylemistake) [reply code/game/machinery/porta_turret/portable_turret.dm:203]: seems pointless, matter of taste

**Review by stylemistake — DISMISSED** [review]

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 3 above)
Comment (ShizCalev) [reply code/game/machinery/porta_turret/portable_turret.dm:203]: if you can do it, do it imo.

Comment (Arkatos1) [reply code/game/machinery/porta_turret/portable_turret.dm]: Done.

**Pull request merged**